pkgs/os-specific/darwin/itsycal/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib
+, fetchurl
+, stdenvNoCC
+, unzip
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "itsycal";
+  version = "0.15.3";
+
+  src = fetchurl {
```

Comment:
Any specific reason why `fetchurl` used but not `fetchzip`?

Reply:
Nope, I didn't realize there was `fetchzip` will change it

Reply:
On a second note maybe not... `fetchzip` is getting different hash and it seems like it's not downloading stuff properly either?

```console
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/ryja8i37q8kci47blxdcmjr6sm51lr9k-source
source root is .
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
no Makefile or custom buildPhase, doing nothing
@nix { "action": "setPhase", "phase": "installPhase" }
installing
total 12
-rw-r--r-- 1 _nixbld1 nixbld  158 Mar 26 21:58 .sandbox.sb
-rw-r--r-- 1 _nixbld1 nixbld 5104 Mar 26 21:58 env-vars
drwxr-xr-x 3 _nixbld1 nixbld   96 Jan  1  1970 source
mv: cannot stat 'Itsycal.app': No such file or directory
```

I suggest it stays `fetchurl` unless `fetchzip` requires extra stuff I'm not aware of that would fix this? 🤔

Reply:
This worked for me (haven't tested the app tho):

```diff
diff --git a/pkgs/os-specific/darwin/itsycal/default.nix b/pkgs/os-specific/darwin/itsycal/default.nix
index 4ce7d3ae084f..8d727e703650 100644
--- a/pkgs/os-specific/darwin/itsycal/default.nix
+++ b/pkgs/os-specific/darwin/itsycal/default.nix
@@ -1,5 +1,5 @@
 { lib
-, fetchurl
+, fetchzip
 , stdenvNoCC
 , unzip
 }:
@@ -8,20 +8,16 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "itsycal";
   version = "0.15.3";

-  src = fetchurl {
+  src = fetchzip {
     url = "https://itsycal.s3.amazonaws.com/Itsycal-${finalAttrs.version}.zip";
-    hash = "sha256-5aJzSuqq31B33jW4lV8vuU3eurpZBoyIW/AOC9/pxng=";
+    hash = "sha256-jpTlJY7yAARrkHzreQKbFaKj0sYp950R0qPPcDeY6AE=";
   };

-  sourceRoot = ".";
-
-  nativeBuildInputs = [ unzip ];
-
   installPhase = ''
     runHook preInstall

-    mkdir -p $out/Applications
-    mv Itsycal.app $out/Applications
+    mkdir -p $out/Applications/Itsycal.app
+    cp -R . $out/Applications/Itsycal.app

     runHook postInstall
   '';
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index c014c268962a..973238d5c8e3 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -38276,6 +38276,8 @@ with pkgs;

   itsx = callPackage ../applications/science/biology/itsx { };

+  itsycal = callPackage ../os-specific/darwin/itsycal { };
+
   iv = callPackage ../applications/science/biology/iv {
     neuron-version = neuron.version;
   };
```

Reply:
I guess the problem was due to

- `sourceRoot = ".";` (wrong location?)
- `mv Itsycal.app $out/Applications` (nix store is read-only).

Reply:
Thanks for the help!